ckanext/cfpb_extrafields/datastore_actions.py

```diff
@@ -0,0 +1,89 @@
+import ckan.plugins.toolkit as tk
+import collections
+ 
+def default_datastore(rid):
+    ''' provide default datastore parameter values and reduce duplication '''
+    defaults_tuple = collections.namedtuple('defaults_tuple', 'context data json_colname')
+    user = tk.get_action('get_site_user')({'ignore_auth': True}, {})
+    context = {'user': user['name']}
+    data = {'resource_id': rid, 'force': 'true',}
+    json_colname = 'jtable'
+    return defaults_tuple(context=context, data=data, json_colname=json_colname)
+
+def create_datastore_json(rid, json_record, title_colname, json_title): # title_colname='name' json_title='datadict'
+    ''' create a new datastore json element with primary key element of title and record '''
+    defaults = default_datastore(rid)
+    context = defaults.context
+    data = defaults.data
+    json_colname = defaults.json_colname
+    data.update({ 
+    'primary_key': [title_colname],
+    'fields': [{'id': title_colname},{'id': json_colname, 'type': 'json'}],
+    'records': [ {title_colname: json_title, json_colname: json_record} ]
+    })
+    try:
+        ds = tk.get_action('datastore_create')(context, data)
+        return 'created datadict!'
```

Comment:
don't return a string, it's not great for error handling and I don't think you're using it anywhere.  Instead, have your except catch the Validation error then throw a more meaningful error message: 
```
except tk.ValidationError, err:
    raise Exception("Error creating the datastore with context {0} and data {1}: {2}".format(context, data, err.error_dict['info'])
```
This will help with debugging and provide a useful error.  Check [the python docs](https://docs.python.org/2/library/exceptions.htmlhttps://docs.python.org/2/library/exceptions.html) for a list of Errors to use, find one that matches if you can.